polls/views.py

```diff
@@ -1,36 +1,70 @@
 from django.views import View
-from django.shortcuts import render
-from django.contrib.auth.mixins import LoginRequiredMixin
+from django.shortcuts import render, redirect
+from django.contrib.auth.mixins import LoginRequiredMixin, UserPassesTestMixin
 from django.views.generic import TemplateView
 from django.utils import timezone
 
+from .models import CurrentBatchEval, CurrentBatchGold, Task, Annotation
+from .utils import batch_selector, present_task_for_user, check_user_work_permission
+
 class IndexView(TemplateView):
     template_name = "polls/index.html"
 
+class HomeView(TemplateView):
+    template_name = "polls/home.html"
+
 class AuthFlowView(LoginRequiredMixin, View):
     template_name = "polls/auth_flow.html"
-    minutes_after_should_rest = 15
-    minutes_after_can_continue = 75
 
     def get(self, request):
-        if not request.user.first_task_of_this_session_performed_at:
-            # if it's user's first time
-            context = {
-                "should_rest": False,
-                "can_continue": False
-                }
-            return render(request, self.template_name, context)
-
-        current_time = timezone.now()
-        time_diff = current_time - request.user.first_task_of_this_session_performed_at
-        time_diff_minutes = time_diff.total_seconds() / 60
-        should_rest = time_diff_minutes > self.minutes_after_should_rest and \
-                      time_diff_minutes < self.minutes_after_can_continue
-        can_continue = time_diff_minutes > self.minutes_after_can_continue
-        rest_time = round(self.minutes_after_can_continue - time_diff_minutes) + 1
+        can_continue, should_rest, rest_time = check_user_work_permission(request.user)
         context = {
-            "should_rest": should_rest,
             "can_continue": can_continue,
+            "should_rest": should_rest,
             "rest_time": rest_time
         }
-        return render(request, self.template_name, context)
\ No newline at end of file
+        return render(request, self.template_name, context)
+
+class TaskFlowView(LoginRequiredMixin, UserPassesTestMixin, View):
+    template_name = "polls/task_flow.html"
+
```

Comment:
Are we currently checking if the user is frozen / banned? If not, just create an issue for that and do it later.

Reply:
Yes, in `test_func` under this class.

Reply:
if locked user tries to access `task-flow` page, they will see 403 (forbidden)

Reply:
Can you post screenshots showing the steps from logging in to seeing the banned page?

Reply:
yes

Reply:
403 and 404 pages can also be customized/styled

Reply:
"
If `user.is_locked`, they are given a screen that says: "Thank you
for contributing." and can do no more work."

That's not a 403 or 404 page. Please create an issue to fix this later.

Reply:
Ok

Reply:
Please make an issue and link it here

Reply:
https://github.com/turian/audio-discrimination-crowdsource/issues/20